homework/index.js

```diff
@@ -29,8 +29,32 @@
     return elem;
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function renderRepoDetails(repo, table) {
+    const RepositoryTr = createAndAppend('tr', table);
+    createAndAppend('th', RepositoryTr, {
+      text: 'Repository:',
+    });
+    const RepoNameTh = createAndAppend('td', RepositoryTr);
+    createAndAppend('a', RepoNameTh, {
+      text: repo.name,
+      href: repo.html_url,
+      target: '_blank',
+    });
+
+    function renderRepoTable(name, path) {
```

Comment:
Avoid embedded named functions like this. Place at the file level (i.e. at the same level as, for instance, the `renderRepoDetails()` function. Of course, you will then need to add an extra parameter: `table`. And I would s

```js

function renderRepoTable(table, name, path) {
  const tr = createAndAppend('tr', table);
  createAndAppend('th', tr, {
    text: name,
  });
  createAndAppend('td', tr, {
    text: path,
  });
  return tr;
}
```

I also added a statement at the end: `return tr`. You can use the return value to handle the exception for the first row that requires a `<a>` instead of simple text content. Your `renderRepoDetails()` function could now look like this:

```js
function renderRepoDetails(repo, table) {
  const firstRow = renderRepoTable(table, 'Repository', '');
  createAndAppend('a', firstRow.lastChild, {
    text: repo.name,
    href: repo.html_url,
    target: '_blank',
  });
  renderRepoTable(table, 'Description:', repo.description);
  renderRepoTable(table, 'Forks:', repo.forks);
  const date = new Date(repo.updated_at).toLocaleString('en-US');
  renderRepoTable(table, 'Updated:', date);
}
```